scripts/init.sh

```diff
@@ -20,34 +23,37 @@ vars_to_check=(
 
 for var in "${vars_to_check[@]}"; do
   check_env_var "${var}"
-  export "${var}"
+  export "${var?}"
 done
 
+# BSD sed?
+sed_i=("-i")
+[ "$(uname)" = "Darwin" ] && sed_i=("-i ''")
+
 # Setting the Node Role
 env_file_exist "${ROOT_DIR}/${ENV_FILE}"
-# shellcheck disable=SC1090
 SCNODE_ROLE="$(grep 'SCNODE_ROLE=' "${ROOT_DIR}/${ENV_FILE}" | cut -d '=' -f2)" || { echo "SCNODE_ROLE value is wrong. Check ${ROOT_DIR}/${ENV_FILE} file"; exit 1; }
 
 if [ -z "${SCNODE_ROLE:-}" ]; then
   read -rp "What kind of node type would you like to run('rpc' or 'forger'): " scnode_role_value
   while [[ ! "${scnode_role_value}" =~ ^(rpc|forger)$  ]]; do
-    echo -e ""Error: Node type can only be 'rpc' or 'forger'. Try again...\n""
+    echo -e "Error: Node type can only be 'rpc' or 'forger'. Try again...\n"
     read -rp "What kind of node type would you like to run('rpc' or 'forger'): " scnode_role_value
   done
-  sed -i'' "s/SCNODE_ROLE=.*/SCNODE_ROLE=${scnode_role_value}/g" ${ROOT_DIR}/${ENV_FILE}
-  if [ ${scnode_role_value} == "forger" ]; then
-    sed -i'' 's/SCNODE_FORGER_ENABLED=.*/SCNODE_FORGER_ENABLED=true/g' ${ROOT_DIR}/${ENV_FILE}
-    sed -i'' 's/SCNODE_FORGER_MAXCONNECTIONS=.*/SCNODE_FORGER_MAXCONNECTIONS=20/g' ${ROOT_DIR}/${ENV_FILE}
-    sed -i'' 's/SCNODE_WS_CLIENT_ENABLED=.*/SCNODE_WS_CLIENT_ENABLED=true/g' ${ROOT_DIR}/${ENV_FILE}
+  sed "${sed_i[@]}" "s/SCNODE_ROLE=.*/SCNODE_ROLE=${scnode_role_value}/g" "${ROOT_DIR}/${ENV_FILE}"
+  if [ "${scnode_role_value}" = "forger" ]; then
+    sed "${sed_i[@]}" 's/SCNODE_FORGER_ENABLED=.*/SCNODE_FORGER_ENABLED=true/g' "${ROOT_DIR}/${ENV_FILE}"
+    sed "${sed_i[@]}" 's/SCNODE_FORGER_MAXCONNECTIONS=.*/SCNODE_FORGER_MAXCONNECTIONS=20/g' "${ROOT_DIR}/${ENV_FILE}"
+    sed "${sed_i[@]}" 's/SCNODE_WS_CLIENT_ENABLED=.*/SCNODE_WS_CLIENT_ENABLED=true/g' "${ROOT_DIR}/${ENV_FILE}"
   else
-    sed -i'' 's/SCNODE_FORGER_ENABLED=.*/SCNODE_FORGER_ENABLED=false/g' ${ROOT_DIR}/${ENV_FILE}
-    sed -i'' 's/SCNODE_WS_CLIENT_ENABLED=.*/SCNODE_WS_CLIENT_ENABLED=false/g' ${ROOT_DIR}/${ENV_FILE}
+    sed "${sed_i[@]}" 's/SCNODE_FORGER_ENABLED=.*/SCNODE_FORGER_ENABLED=false/g' "${ROOT_DIR}/${ENV_FILE}"
+    sed "${sed_i[@]}" 's/SCNODE_WS_CLIENT_ENABLED=.*/SCNODE_WS_CLIENT_ENABLED=false/g' "${ROOT_DIR}/${ENV_FILE}"
   fi
 fi
 
 # Checking if .env file exist and sourcing
 env_file_exist "${ROOT_DIR}/${ENV_FILE}"
-# shellcheck disable=SC1090
+# shellcheck source=../.env.template.mainnet.eon
```

Comment:
this could also be gobi right?

Reply:
It could, yes. Was just need to quiet shellcheck.